crypto/bn/bn_prime.c

```diff
@@ -49,15 +49,91 @@ int BN_GENCB_call(BN_GENCB *cb, int a, int b)
     return 0;
 }
 
-int BN_generate_prime_ex(BIGNUM *ret, int bits, int safe,
-                         const BIGNUM *add, const BIGNUM *rem, BN_GENCB *cb)
+/*
+ * This defines the security level |sec_level| in bits of doing |mr_checks|
+ * Miller-Rabin on a random prime of |prime_size|.
+ *
+ * The table is sorted by |mr_checks|, then |prime_size|. This is so that we
+ * find the minimum number of Miller-Rabin for a given |prime_size| and
+ * |sec_level|.
+ *
+ * This table is generated using the algorithm of FIPS 186-4 section F.1
+ */
+static const struct {
+    int prime_size;
+    int mr_checks;
+    int sec_level; } prime_mr_sec_level[] = {
```

Comment:
This looks wrong to me: put the `}` on the next line

Reply:
I really had no idea how to format this.

Reply:
introduce a typedef and separate the type from the variable.

Reply:
@richsalz, imho it is unnecessary to typedef a struct which is used only once for a single array. So I think it is enought to follow @mattcaswell's suggestion.

Reply:
I was just offering a way to do it.